src/clockwork/amqp.clj

```diff
@@ -30,4 +41,19 @@
     (log/info (format "[amqp/connect] [%s]" (config/amqp-uri)))
     (declare-exchange channel exchange-cfg)
     (declare-queue channel exchange-cfg queue-cfg (keys handlers))
-    (lc/blocking-subscribe channel (:name queue-cfg) (partial message-router handlers))))
+    (lc/blocking-subscribe channel (:name queue-cfg) (partial message-router handlers))
+    channel))
+
+(defn publish-msg
+  [routing-key msg]
+  (try+
+    (let [timeNow (new java.util.Date)]
+      (lb/publish (channel)
+                  (config/exchange-name)
+                  routing-key
+                  (cheshire/encode {:message      msg
+                                    :timestamp_ms (.getTime timeNow)})
+                  {:content-type "application/json"
+                   :timestamp    timeNow}))
```

Comment:
Possibly nice for tracking, but we don't require the timestamp in infosquito, and we definitely don't care about the content-type since at present we send an empty message. Not sure if you're planning to change that, or if this code is mostly for tracking/monitoring purposes.